lib/js/src/manager/file/BaseFileManager.js

```diff
@@ -0,0 +1,245 @@
+        this._remoteFiles = [];
+
+        const rpcRequest = new RpcRequest(
+            {
+                functionName: 'ListFiles',
+                parameters: {},
+                rpcType: RpcType.REQUEST,
+            }
+        );
+
+        const rpcResponse = await this._internalInterface.sendRPC(rpcRequest);
+        if (rpcResponse.getSuccess()) {
+            this._bytesAvailable = rpcResponse.getSpaceAvailable() ? rpcResponse.getSpaceAvailable() : SPACE_AVAILABLE_MAX_VALUE;
+            const filenames = rpcResponse.getFilenames();
+            if (Array.isArray(filenames)) {
+                this._remoteFiles = filenames;
+            }
+            // on callback set manager to ready state
+            return this._transitionToState(BaseSubManager.READY);
+        } else {
+            this._bytesAvailable = SPACE_AVAILABLE_MAX_VALUE;
+            return this._transitionToState(BaseSubManager.READY);
+        }
+    }
+
+    /**
+	 * Attempts to delete the desired file from core, calls listener with indication of success/failure
+	 * @param {String} fileName name of file to be deleted
+	 * @param {CompletionListener} listener callback that is called on response from core
+	 */
+    async deleteRemoteFileWithName (fileName, onComplete = null) {
+        const rpcRequest = new RpcRequest(
+            {
+                functionName: 'DeleteFile',
+                parameters: {
+                    syncFileName: fileName,
+                },
+                rpcType: RpcType.REQUEST,
+            }
+        );
+
+        const rpcResponse = await this._internalInterface.sendRPC(rpcRequest);
+        const isSuccess = rpcResponse.getSuccess();
+        if (isSuccess) {
+            this._bytesAvailable = rpcResponse.getSpaceAvailable() ? rpcResponse.getSpaceAvailable() : SPACE_AVAILABLE_MAX_VALUE;
+            this._remoteFiles.splice(this._remoteFiles.indexOf(fileName), 1);
+            this._uploadedEphemeralFileNames.splice(this._remoteFiles.indexOf(fileName), 1);
+        }
+        if (onComplete !== null) {
+            onComplete(isSuccess);
+        }
+        return isSuccess;
+    }
+
+
+    /**
+	 * Attempts to delete a list of files from core, calls onComplete with indication of success/failure.
+	 * @param fileNames list of file names to be deleted
+	 * @param onComplete callback that is called once core responds to all deletion requests
+	 */
+    async deleteRemoteFilesWithNames (fileNames, onComplete = null) {
+        const results = [];
+
+        if (fileNames.length > 0) {
+            for (const filename of fileNames) {
+                results.push(this.deleteRemoteFileWithName(filename));
+            }
+            for (const idx in results) {
+                results[idx] = await Promise.resolve(results[idx]);
+            }
+        }
+
+        if (onComplete !== null) {
+            onComplete(results);
+        }
+        return results;
+    }
+
+    async uploadFile (file, onComplete = null) {
```

Comment:
Method missing JSDoc documentation. Also recommend renaming `file` to `sdlFile` for clarity.